constraints/src/main/java/gov/nasa/jpl/aerie/constraints/model/Violation.java

```diff
@@ -1,28 +1,37 @@
 package gov.nasa.jpl.aerie.constraints.model;
 
+import gov.nasa.jpl.aerie.constraints.time.Interval;
 import gov.nasa.jpl.aerie.constraints.time.Windows;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.stream.StreamSupport;
 
 public final class Violation {
   public final List<Long> activityInstanceIds;
   public final List<String> resourceNames;
-  public final Windows violationWindows;
+  public final List<Interval> violationWindows;
 
   public Violation(final List<Long> activityInstanceIds, final List<String> resourceNames, final Windows violationWindows) {
     this.activityInstanceIds = new ArrayList<>(activityInstanceIds);
     this.resourceNames = new ArrayList<>(resourceNames);
-    this.violationWindows = violationWindows;
+    this.violationWindows = new ArrayList<>(violationWindows.size());
+    for (final var interval: violationWindows.iterateEqualTo(true)) this.violationWindows.add(interval);
   }
 
   public Violation(final Windows violationWindows) {
     this(new ArrayList<>(), new ArrayList<>(), violationWindows);
   }
 
+  public Violation(final List<Long> activityInstanceIds, final List<String> resourceNames, final List<Interval> intervals) {
+    this.activityInstanceIds = activityInstanceIds;
```

Comment:
This constructor needs to clone `activityInstanceIds` at the very least, or else two Violation objects sharing the same activityInstanceIds will affect each other when `addActivityId` is called on one of them.

Reply:
I thought the standard was that for constructors where you provide the fields, you assume it takes ownership of the data.

Reply:
I'm used to a [defensive copy](http://www.javapractices.com/topic/TopicAction.do?Id=15) convention myself. Defensive copying needs no ongoing mindfulness to stay correct, whereas ownership passing requires every caller to know about the ownership expectations and manually check that what they're doing is safe. For me, such a convention would deserve a `SAFETY` comment on every call site explaining why it's safe to make that call.

I thought the use of this constructor in `GetSimulationResultsAction` might constitute a bug, but on examination I don't think the originating list lives past the newly constructed `Violation` object? Still, it's not *immediately* obvious, and as code shifts that invariant could easily get lost.

Reply:
Alright

Reply:
Oh I forgot that the other constructor copies as well.